code/modules/mob/new_player/new_player.dm

```diff
@@ -40,6 +41,35 @@
 	privacy_consent()
 	return FALSE
 
+/mob/new_player/proc/challenge_phrase()
+	if(!config.challenge_phrase) //check the config if there is in fact a challenge phrase, if not then just return true
+		return TRUE
+	establish_db_connection()
+	if(!dbcon.IsConnected())
+		return TRUE
+	var/DBQuery/query = dbcon.NewQuery("SELECT challenge FROM [format_table_name("ban")] WHERE ckey = '[src.ckey]' AND challenge=1")
+	query.Execute()
+	while(query.NextRow()) //Find a row that matches ckey where challenge is requested
+		challenge_phrase = TRUE // set var to TRUE to disallow other functions and recalling the original new_player_panel() if they fail
+		//sry for this input mess
+		var/given_phrase
+		var/ask_phrase = input(src, "The conditions of your unban require you to provide the hidden phrase within the rules to prove you have read them fully. You must complete this to join. Please type the phrase below.", "Administrative Code Phrase", given_phrase) as text //ask for the phrase
+		given_phrase = ask_phrase
+		if(cmptext(sanitize(given_phrase), config.challenge_phrase)) //see if the phrase matches the one in the config - cmptext makes it not caps sensitive
+			challenge_phrase = FALSE //remove the var lock on joining things
+			//create new thing to remove the phrase
+			var/DBQuery/new_query = dbcon.NewQuery("UPDATE [format_table_name("ban")] SET challenge = 0  WHERE ckey = '[src.ckey]'")
+			new_query.Execute()
+			to_chat(src, "<span class='danger'>Correct phrase. Functions unlocked. Welcome back to Paradise.</span>") //tell user they won
+			return TRUE //allow for new_player_panel to continue 
+		else
+			to_chat(src, "<span class='danger'>Incorrect phrase. Please completely read all of the rules, it is a set of words that let us know you have read the rules in full.</span>") //tell user to redo
+			spawn(20) new_player_panel() //recall the thing so they arent in limbo with delay of 2 seconds so no DB overwhelm

```

Comment:
This needs to use a timer, not a spawn

Reply:
Why not?

Reply:
Hmm, actually that's a very good question. #8946 leads me to believe you should favour addtimer over spawn. @Fox-McCloud may be able to explain this better than me 😄